{{cookiecutter.project_slug}}/.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,38 @@
+name: Run pre-commit in blueprint
+
+on:
+  push:
+    branches:
+    - dev_refactor
+  pull_request:
+    branches:
+    - dev_refactor
+
+jobs:
+  blueprint-pre-commit:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9.7
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        miniconda-version: "latest"
+        channels: conda-forge
+        channel-priority: flexible
+        show-channel-urls: true
+    - name: Create dev env from unpinned reqs
+      run: |
+        conda env create --name dev_blueprint --file requirements/requirements.yml
+        conda env update --name dev_blueprint --file requirements/dev-requirements.yml
+    - name: Install pre-commit hooks
+      run: |
+        conda run --name dev_blueprint pre-commit install-hooks
+    - name: Run pre-commit hooks
+      run: |
+        conda run --name dev_blueprint pre-commit run --all-files
```

Comment:
comment from discussion with Daniel, we had a discussion about what is tested where (jenkins github actions). At the beginning github actions where testing only precommit, but the tests were exclusively tested by jenkins. But we checked the quotas and it seems that even for the free plan the limit of github actions is quite generous, so that we can move tests inside github actions. This PR is doing so for the new-project.yml but to be consistent with the new-project.yml, the new pattern would be like we run precommit & test in github actions for unpinned environment, while we reserve jenkins for production tests of pinned environment. Therefore to be consistent we should also run tests here. And we can remove then unpinned environment test from jenkins.

Reply:
Good news with the Actions quotas. Makes sense to use Jenkins with pinned deps.

Reply:
Left it as is for now, pending a decision as to what tests belong where.